File: Leetcode/2053/2053.go
type Node struct {
	freq, index int
}
type Pair struct {
	s string
	index int
}
func kthDistinct(arr []string, k int) string {
	m := make(map[string]Node)
	for i := 0; i < len(arr); i++ {
		val, ok := m[arr[i]]
		if !ok {
			m[arr[i]] = Node{
				freq: 1,
				index: i,
			}
		} else {
			m[arr[i]] = Node{
				freq: val.freq+1,
				index: i,
			}
		}
	}
	var temp []Pair
	for k, v := range m {
		if v.freq == 1 {
			temp = append(temp, Pair{
				s: k,
				index: v.index,
			})
		}
	}
	if len(temp) < k {
		return ""
	}
	sort.Slice(temp, func(i, j int) bool {
		return temp[i].index < temp[j].index
	})
	return temp[k-1].s
}

File: Leetcode/2053/2053.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string,pair<int,int>> unmap;
        int n = (int)arr.size();
        for (int i = 0; i < n; ++i) {
            ++unmap[arr[i]].first;
            unmap[arr[i]].second = i;
        }
        vector<pair<string,int>> vec;
        for (auto &[k,v] : unmap) {
            if (v.first == 1) {
                vec.emplace_back(k, v.second);
            }
        }
        if ((int)vec.size() < k) {
            return string();
        }
        sort(vec.begin(),vec.end(),[](const auto &l, const auto &r) {
            return l.second < r.second; 
        });
        return vec[k-1].first;
    }
};